abba/json/src/Parse.cpp

```diff
@@ -0,0 +1,480 @@
+#include "Parse.hpp"
+
+Parse::Parse(std::string filename) : json(filename)
+{
+
+}
+
+void Parse::parse()
+{
+	_file = &json.getFile();
+	std::cout << "i'm here" << std::endl;
+	getToken();
+	Json *root = new Json();
+	root->type = Json::OBJECT;
+	root = parseObject(root);
+	std::cout << "DONE PARSING!!!!!!!!!!!!" << std::endl;
+}
+
+char Parse::getWithoutWhiteSpace()
+{
+	std::cout << "getWithoutWhiteSpace function called" << std::endl;
+	char	c = ' ';
+	while (c == ' ' || c == '\t' || c == '\n' || \
+			c == '\v' || c == '\f' || c == '\r')
+	{
+		_file->get(c);
+		if (!_file)
+			return c;
+	}
+	return c;
+}
+
+Parse::Type Parse::getToken()
+{
+	char		c;
+	Type	type;
+
+//	std::cout << "getToken function called" << std::endl;
+	if (_file->eof())
+	{
+		std::cout << "finish reading" << std::endl;
+	}
+	_prevPos = _file->tellg();
+	c = getWithoutWhiteSpace();
+	if (c == '"')
+	{
+		type.type = STRING;
+		_file->get(c);
+		while (c != '\n' && c != '"')
```

Comment:
A newline value in a JSON string would be part of the string itself, the value can for example become `"ab\nen"` which is all part of the same string